### DISCUSSION TITLE
invoke never be called when using useMachine

### DISCUSSION THREAD

**Discussion by ro7584** [statelyai/xstate#2593] [Q&A] [answered]
Hi everyone,

I'm facing a weird problem when invoking service in machine with react `useMachine`.

I was expected that invoke:src should be called but it didn't.
The console shows that
> Warning: No service found for invocation 'user.loading:invocation[0]' in machine 'user'. 
    in App 

This problem can be reproduced in 
https://codesandbox.io/s/xstate-react-service-not-found-b95yu?file=/src/App.js:262-287

I also try to use interpret directly, and it works.
```javascript
const service = interpret(machine).start();
service.send("FETCH");
```

Not sure what I'm missing. Any suggestion will be appreciated.

**Comment by Andarist** [comment, marked as answer]
This has been a bug in `@xstate/react@1.6.0` but it has been fixed in [`1.6.1`](https://github.com/statelyai/xstate/releases/tag/%40xstate%2Freact%401.6.1):
https://codesandbox.io/s/xstate-react-service-not-found-forked-ymcn6?file=/src/App.js

**Reply by ro7584** [reply]
wow. thank you for fixed this bug so quickly ❤️
